public/js/graph/react_node.js.jsx

```diff
@@ -174,93 +212,223 @@ var ReactRegion = React.createClass({
+    getInitialState: function () {
+        return {
+            status: this.props.parents.length === 0 ? 'takeable' : 'inactive',
+        };
+    },
+    
+    isSelected: function () {
+        return this.state.status === 'active' || this.state.status === 'overridden';
+    },
+    
+    arePrereqsSatisfied: function (svg) {
+        function isAllTrue(element, index, array) {
+            return svg.refs['nodes'].refs[element] ? svg.refs['nodes'].refs[element].isSelected() :
+                                                     svg.refs['bools'].refs[element].isSelected();
+        }
         
-        $('.path').map(function(key, element) {
-            if (id == element.getAttribute('data-target-node')){
-                parents.push(element.getAttribute('data-source-node'));
-                inEdges.push(element.id);
+        if (this.props.logicalType === 'AND') {
+            return this.props.parents.every(isAllTrue);     
+        } else if (this.props.logicalType === 'OR') {
+            return this.props.parents.some(isAllTrue);
+        }
+    },
+    
+    updateNode: function (svg) {
+        if (this.arePrereqsSatisfied(svg)) {
+            if (this.isSelected() || this.props.hybrid) {
+                this.setState({status: 'active'});
+            } else {
+                this.setState({status: 'takeable'});
             }
-            if (id == element.getAttribute('data-source-node')){
-                children.push(element.getAttribute('data-target-node'));
-                outEdges.push(element.id);
+        } else {
+            if (this.isSelected() && !this.props.hybrid) {
+                this.setState({status: 'overridden'});
+            } else {
+                this.setState({status: 'inactive'});
             }
-        });
-        
-        if (parents.length == 0){
-            status = 'takeable';
         }
-        
-        return {
-            id: id,
-            parents: parents,
-            children: children,
-            outEdges: outEdges,
-            inEdges: inEdges,
-            logicalType: type,
-            updated: false,
-            hybrid: this.props.hybrid,
-            status: status
-        };
+
+        // Always update children of hybrids
+        if (this.props.hybrid) {
+            $.each(this.props.childs, function (i, node) {
+                var currentNode = svg.refs['nodes'].refs[node] ? svg.refs['nodes'].refs[node] :
+                                                                 svg.refs['bools'].refs[node];
+                currentNode.forceUpdate(function () { currentNode.updateNode(svg) });
+            });
+            $.each(this.props.outEdges, function (i, edge) {
+                var currentEdge = svg.refs['edges'].refs[edge];
+                currentEdge.forceUpdate(function () { currentEdge.updateEdge(svg) });
+            });
+        }
     },
     
-    handleClick: function(event) {
-        //code here
+    turn: function (svg) {
```

Comment:
The name `turn` gives me the impression that this function rotates a node or something.